include/rbs/location.h

```diff
@@ -32,6 +32,18 @@ typedef struct rbs_location {
     rbs_loc_children *children;
 } rbs_location_t;
 
+typedef struct rbs_location_list_node {
+  rbs_location_t *loc;
+  struct rbs_location_list_node *next;
+} rbs_location_list_node_t;
+
+typedef struct rbs_location_list {
+  rbs_allocator_t *allocator;
+  rbs_location_list_node_t *head;
+  rbs_location_list_node_t *tail;
+  size_t length;
+} rbs_location_list_t;
```

Comment:
The new `MethodTypesAnnotation` type node has an attribute called `vertical_bar_locations` that holds a list of locations. Since we currently don't have a data structure for that, I added this by mimicking the `rbs_node_list` struct.